test/integration/navsat.cc

```diff
@@ -0,0 +1,332 @@
+/*
+ * Copyright (C) 2021 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+*/
+
+#include <gtest/gtest.h>
+
+#include <sdf/sdf.hh>
+
+#include <ignition/math/Helpers.hh>
+#include <ignition/sensors/NavSatSensor.hh>
+#include <ignition/sensors/SensorFactory.hh>
+
+#include "test_config.h"  // NOLINT(build/include)
+#include "TransportTestTools.hh"
+
+using namespace ignition;
+using namespace sensors;
+using namespace std::chrono_literals;
```

Comment:
I think it would be best to avoid using namespaces that aren't unique to the project (in this case, ignition sensors) in order to avoid potential naming collisions later on

Reply:
That's a good point in general, but `std::chrono_literals` is a very convenient namespace to have as it allows us to use time literals (i.e. `1s` instead of the verbose alternative `std::chrono::seconds(1)`) which are way easier to read. It's such a nice-to-have that I think is worth the namespace, what do you think?

Reply:
I think it's okay to leave it in here since it's a test file (which means users probably aren't linking against this), and since it avoids the long verbose alternative. Moving forward though, I think it could be worth updating our style guide to mention when including other namespaces is allowed or not. My initial impression is that we should follow something along these lines:
1. **never** use another ignition project namespace in an ignition project file (example: for a file in the `ign-gazebo` project, don't use a namespace like `ignition::sensors`)
2. The source files of an ignition project should only use the same ignition project's namespace (i.e., no namespaces from the C++ stdlib should be used)
3. For test files of an ignition project, you should try to only use the same ignition project's namespace, but are allowed to use namespaces from things like the C++ stdlib if doing so makes code easier to read (example: you can use `using namespace std::chrono_literals` so that you can write `1s` instead of `std::chrono::seconds(1)`)

If you think it would be worth documenting do's and dont's of namespaces, we can open a separate issue/PR in the appropriate repository and continue the discussion there.

Reply:
These look reasonable to me. I also think we should explicitly disallow namespace aliases because they make the code harder to read (i.e. `namespace ir = ignition::rendering;`). I think some of these are already documented at [Google's style guide](https://google.github.io/styleguide/cppguide.html#Namespaces). +1 to document Ignition-specific rules on [our style guide](https://ignitionrobotics.org/docs/all/contributing#style-guides).

Reply:
> I also think we should explicitly disallow namespace aliases because they make the code harder to read

Agreed :+1: 

> +1 to document Ignition-specific rules on our style guide.

Would you be able to open a PR that adds this, or should an issue be created for now so that you or I can address this later when we have the time?

Reply:
+1 for the issue, thanks! :bowing_woman:

Reply:
issue has been created: https://github.com/ignitionrobotics/docs/issues/225

I'll mark this as resolved